Source/Core/Core/ConfigLoaders/GameConfigLoader.cpp

```diff
@@ -0,0 +1,415 @@
+  return {"", ""};
+}
+
+// INI Game layer configuration loader
+class INIGameConfigLayerLoader : public Config::ConfigLayerLoader
+{
+public:
+  INIGameConfigLayerLoader(const std::string& id, u16 revision, bool global)
+      : ConfigLayerLoader(global ? Config::LayerType::GlobalGame : Config::LayerType::LocalGame),
+        m_id(id), m_revision(revision)
+  {
+  }
+
+  void Load(Config::Layer* config_layer) override
+  {
+    IniFile ini;
+    if (config_layer->GetLayer() == Config::LayerType::GlobalGame)
+    {
+      for (const std::string& filename : GetGameIniFilenames(m_id, m_revision))
+        ini.Load(File::GetSysDirectory() + GAMESETTINGS_DIR DIR_SEP + filename, true);
+    }
+    else
+    {
+      for (const std::string& filename : GetGameIniFilenames(m_id, m_revision))
+        ini.Load(File::GetUserPath(D_GAMESETTINGS_IDX) + filename, true);
+    }
+
+    const std::list<IniFile::Section>& system_sections = ini.GetSections();
+
+    for (auto section : system_sections)
+    {
+      const std::string section_name = section.GetName();
+      if (section.HasLines())
+      {
+        // Trash INI File chunks
+        std::vector<std::string> chunk;
+        section.GetLines(&chunk, true);
+
+        if (chunk.size())
+        {
+          const auto mapped_config = MapINIToRealLocation(section_name, "");
+
+          if (mapped_config.section.empty() && mapped_config.key.empty())
+            continue;
+
+          auto* config_section =
+              config_layer->GetOrCreateSection(mapped_config.system, mapped_config.section);
+          config_section->SetLines(chunk);
+        }
+      }
+
+      // Regular key,value pairs
+      const IniFile::Section::SectionMap& section_map = section.GetValues();
+
+      for (auto& value : section_map)
+      {
+        const auto mapped_config = MapINIToRealLocation(section_name, value.first);
+
+        if (mapped_config.section.empty() && mapped_config.key.empty())
+          continue;
+
+        auto* config_section =
+            config_layer->GetOrCreateSection(mapped_config.system, mapped_config.section);
+        config_section->Set(mapped_config.key, value.second);
+      }
+    }
+
+    // Game INIs can have controller profiles embedded in to them
+    std::string num[] = {"1", "2", "3", "4"};
+
+    if (m_id != "00000000")
+    {
+      std::tuple<std::string, std::string, Config::System> profile_info[] = {
+          std::make_tuple("Pad", "GCPad", Config::System::GCPad),
+          std::make_tuple("Wiimote", "Wiimote", Config::System::WiiPad),
+      };
+
+      for (auto& use_data : profile_info)
+      {
+        std::string type = std::get<0>(use_data);
+        std::string path = "Profiles/" + std::get<1>(use_data) + "/";
+
+        Config::Section* control_section =
+            config_layer->GetOrCreateSection(std::get<2>(use_data), "Controls");
+
+        for (int i = 0; i < 4; i++)
```

Comment:
This can just call `num.size()` once `num is made into a `std::array`, or be a ranged-for loop.